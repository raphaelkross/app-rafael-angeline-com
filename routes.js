/* eslint-env node */

// eslint-disable-next-line no-undef
const routes = (module.exports = require("next-routes")());

routes
	.add("blog", "/blog/:page?")
	.add("category", "/category/:slug/:page?")
	.add("post", "/:slug", "index");
