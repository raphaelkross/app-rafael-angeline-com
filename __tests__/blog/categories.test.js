/* eslint-env jest */

import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import Categories from "../../components/blog/categories";
import { Link } from "../../routes";

describe("Categories Widget", () => {
	it("should match complete layout snapshot", () => {
		const cats = [
			{ href: "#1", label: "Category 1" },
			{ href: "#2", label: "Category 2" },
			{ href: "#3", label: "Category 3" }
		];

		const component = renderer.create(<Categories items={cats} />);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("should not render if items is empty or not passed", () => {
		const cats = [];
		let widget = shallow(<Categories items={cats} />);

		expect(widget.find(".widget").length).toEqual(0);

		widget = shallow(<Categories />);

		expect(widget.find(".widget").length).toEqual(0);
	});

	it("should render proper number of items as links", () => {
		const cats = [
			{ href: "#1", label: "Category 1" },
			{ href: "#2", label: "Category 2" },
			{ href: "#3", label: "Category 3" }
		];

		let widget = shallow(<Categories items={cats} />);

		const links = widget.find(Link);

		expect(links.length).toEqual(3);
	});

	it("should render links with proper href and label", () => {
		const cats = [{ href: "#1", label: "Category 1" }];

		let widget = shallow(<Categories items={cats} />);

		const link = widget.find(Link).at(0);

		expect(link.prop("route")).toEqual("/#1");
		expect(link.children("a").prop("children")).toEqual("Category 1");
	});
});
