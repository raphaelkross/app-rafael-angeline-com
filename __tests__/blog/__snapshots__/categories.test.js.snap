// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Categories Widget should match complete layout snapshot 1`] = `
<div
  className="jsx-4220160363 widget"
>
  <h3
    className="jsx-4220160363"
  >
    Categories.
  </h3>
  <ul
    className="jsx-4220160363"
  >
    <li
      className="jsx-4220160363"
    >
      <a
        className="jsx-4220160363"
        href="http://localhost:3000/#1"
        onClick={[Function]}
      >
        Category 1
      </a>
    </li>
    <li
      className="jsx-4220160363"
    >
      <a
        className="jsx-4220160363"
        href="http://localhost:3000/#2"
        onClick={[Function]}
      >
        Category 2
      </a>
    </li>
    <li
      className="jsx-4220160363"
    >
      <a
        className="jsx-4220160363"
        href="http://localhost:3000/#3"
        onClick={[Function]}
      >
        Category 3
      </a>
    </li>
  </ul>
</div>
`;
